src/general/graph_coloring.rs

```diff
@@ -0,0 +1,179 @@
+use crate::data_structures::{Graph, UndirectedGraph};
```

Comment:
question: Why do we import `Graph`?

Reply:
@alexfertel sorry for delayed response, I was in army all this time.

we import `Graph` because some method are implemented in `Graph` (such as `nodes`, `neighbours`) and compiler doesn't happy without this import.

I am not sure how to fix this except of import `Graph`.

![image](https://github.com/alexfertel/rust-algorithms/assets/1453981/236a8d00-3a59-4928-a497-5d2e2ccc298d)

Reply:
> sorry for delayed response, I was in army all this time.

No worries at all! Thanks for defending Ukraine 🇺🇦 ❤️ 

> I am not sure how to fix this except of import Graph.

No worries, it makes sense 👍 .